addons/stock/models/stock_orderpoint.py

```diff
@@ -505,11 +508,21 @@ def _procure_orderpoint_confirm(self, use_new_cursor=False, company_id=None, rai
                         origin = '%s - %s' % (orderpoint.display_name, ','.join(origins))
                     else:
                         origin = orderpoint.name
-                    if float_compare(orderpoint.qty_to_order, 0.0, precision_rounding=orderpoint.product_uom.rounding) == 1:
+                    qty_to_order = orderpoint.qty_to_order
+                    product_rounding = orderpoint.product_uom.rounding
+                    if self._context.get('force_to_max') and float_compare(qty_to_order, 0.0, precision_rounding=product_rounding) == 0:
+                        product_context = orderpoint._get_product_context(visibility_days=orderpoint.visibility_days)
+                        virtual_available = orderpoint.product_id.with_context(product_context).read(['virtual_available'])[0]['virtual_available']
+                        qty_forecast_with_visibility = virtual_available + orderpoint._quantity_in_progress()[orderpoint.id]
+                        qty_to_order = orderpoint.product_max_qty - qty_forecast_with_visibility
+                        remainder = orderpoint.qty_multiple > 0 and qty_to_order % orderpoint.qty_multiple or 0.0
+                        if float_compare(remainder, 0.0, precision_rounding=product_rounding) > 0:
+                            qty_to_order += orderpoint.qty_multiple - remainder
```

Comment:
Instead of duplicate the `_compute_qty_to_order` code, you can extract the part defining the `qty_to_order` and use it in the compute and here.

Reply:
Why you need all that? You could just put qty_to_order to be equals to max - qty forecast